numpy/core/numeric.py

```diff
@@ -127,12 +127,22 @@ def zeros_like(a, dtype=None, order='K', subok=True):
         If True, then the newly created array will use the sub-class
         type of 'a', otherwise it will be a base-class array. Defaults
         to True.
+    shape : int or sequence of ints, optional.
+        Overrides the shape of the result.
+
+        .. versionadded:: 1.17.0
 
     Returns
     -------
     out : ndarray
         Array of zeros with the same shape and type as `a`.
 
+    Raises
+    ------
+    ValueError
+        If len(shape) different from a.ndim and order is 'K', or
+        a is not a C/F-layout array
```

Comment:
What does `C/F-layout` mean here?

Reply:
I assume you mean the `order` argument has a invalid value? If so, maybe
```
        If `order` has an invalid value or `order` is 'K' and ``len(shape)``
        is different from ``a.ndim``.
```

Reply:
I think what you mean with your suggestion is that `order` could be any random value different than `CFAK`, right? If that's the case, that's not what I my intent with the documentation, what I mean is that there's no particular memory layout, maybe I got the terminology wrong?

Reply:
And by _no particular memory layout_, I mean something like the result of advanced indexing, for example, stealing that terminology from [indexing documentation](https://docs.scipy.org/doc/numpy/reference/arrays.indexing.html#detailed-notes):

_The memory layout of an advanced indexing result is optimized for each indexing operation and no particular memory order can be assumed._

Reply:
OK, I see what you mean. That is difficult to express exactly, maybe just brute force it: 
```
        If `order` is 'K'  and any of the following is true

            - ``len(shape) != a.ndim``,
            - `a` is neither C nor F contiguous.
```

Reply:
Note that strictly speaking, it should be possible to add 1's to the shape without affecting the contiguity.
```
In [1]: a = ones((2,2), order='C')

In [2]: a.flags.c_contiguous
Out[2]: True

In [3]: a[None,:,:].flags.c_contiguous
Out[3]: True

In [4]: a = ones((2,2), order='F')

In [5]: a.flags.f_contiguous
Out[5]: True

In [6]: a[None,:,:].flags.f_contiguous
Out[6]: True
```
So possibly the correct check would be that the number of number of dimensions > 1 must match. @seberg Thoughts?

Reply:
You're right, I can do that modification if necessary, but I'm not sure this is of utmost importance. Particularly, for our use cases I don't think that affects us.